File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/ControllerDataModel.java
package com.wayfair.brickkitdemo;

import com.wayfair.brickkit.brick.DataModel;

/**
 * Data model for a controller brick.
 */
public class ControllerDataModel extends DataModel {
    private int value;

    /**
     * Constructor with a starting value.
     *
     * @param value the starting value
     */
    public ControllerDataModel(int value) {
        this.value = value;
    }

    /**
     * Change the value.
     *
     * @param value the new value
     */
    public void setValue(int value) {
        this.value = value;
        notifyChange();
    }

    /**
     * Get the current value.
     *
     * @return the value
     */
    public int getValue() {
        return this.value;
    }
}

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/addremove/datamodel/ControllerDataModel.kt
/*
 * Copyright © 2017-2021 Wayfair. All rights reserved.
 */
package com.wayfair.brickkitdemo.addremove.datamodel

import com.wayfair.brickkit.brick.DataModel

/**
 * Data model for a controller brick.
 */
class ControllerDataModel(initialValue: Int) : DataModel() {

    var value: Int = initialValue
        set(value) {
            field = value
            notifyChange()
        }
}
